### PR TITLE
Deprecate RedshiftDataOperatorAsync, RedshiftClusterSensorAsync, S3KeySensorAsync, BatchOperatorAsync, SageMakerProcessingOperatorAsync, SageMakerTransformOperatorAsync

### PR DESCRIPTION
**Pull request by Lee-W**
Deprecate `RedshiftDataOperatorAsync`, `RedshiftClusterSensorAsync`, `S3KeySensorAsync`, `BatchOperatorAsync`, `SageMakerProcessingOperatorAsync`, `SageMakerTransformOperatorAsync` and proxy them to their Airflow OSS provider's counterpart

### CODE REVIEW

**Comment by Lee-W** [comment]
image

**Comment by Lee-W** [comment]
S3
image

**Comment by Lee-W** [comment]
SageMakerProcessingOperatorAsync, SageMakerTransformOperatorAsync

image

**Comment by Lee-W** [comment]
RedshiftClusterSensorAsync

image

File: astronomer/providers/amazon/aws/operators/redshift_data.py
Comment (pankajastro) [inline astronomer/providers/amazon/aws/operators/redshift_data.py:29]: do we need this line? 
looks like `poll_interval ` is kwargs we can just remove this from the init

File: astronomer/providers/amazon/aws/sensors/redshift_cluster.py
Comment (pankajastro) [inline astronomer/providers/amazon/aws/sensors/redshift_cluster.py:22]: shall we remove `poll_interval` from init and check kwargs["poll_interval"] in L21 this way code editor will probably not suggest this param

```python
if kwargs.get("poll_interval"):
    kwargs["poke_interval"] = kwargs["poll_interval"]
```

File: setup.cfg
Comment (pankajastro) [inline setup.cfg:121]: Note: please remove rc before merging it

File: astronomer/providers/amazon/aws/sensors/redshift_cluster.py
Comment (Lee-W) [reply astronomer/providers/amazon/aws/sensors/redshift_cluster.py]: good idea! just changed

File: astronomer/providers/amazon/aws/hooks/sagemaker.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/hooks/sagemaker.py:29]:
```suggestion
    Use :class: `~airflow.providers.amazon.aws.hooks.sagemaker.SageMakerHook` instead
```

File: astronomer/providers/amazon/aws/hooks/sagemaker.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/hooks/sagemaker.py:38]:
```suggestion
                "Please use `airflow.providers.amazon.aws.hooks.sagemaker.SageMakerHook`"
```

File: astronomer/providers/amazon/aws/operators/sagemaker.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:55]:
```suggestion
    Please use :class: `~airflow.providers.amazon.aws.operators.sagemaker.SageMakerTransformOperator`
```

File: astronomer/providers/amazon/aws/operators/sagemaker.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:63]:
```suggestion
                "Please use `airflow.providers.amazon.aws.operators.sagemaker.SageMakerTransformOperator`"
```

File: astronomer/providers/amazon/aws/sensors/redshift_cluster.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/sensors/redshift_cluster.py:29]:
```suggestion
                "This module is deprecated."
```
I guess we won't remove the sensors and operators?

File: astronomer/providers/amazon/aws/operators/batch.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/batch.py:28]:
```suggestion
                "This module is deprecated."
```

File: astronomer/providers/amazon/aws/operators/redshift_data.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/redshift_data.py:20]:
```suggestion
                "This module is deprecated."
```

File: astronomer/providers/amazon/aws/operators/sagemaker.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:42]:
```suggestion
                "This module is deprecated."
```

File: astronomer/providers/amazon/aws/operators/sagemaker.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/operators/sagemaker.py:62]:
```suggestion
                "This module is deprecated."
```

File: astronomer/providers/amazon/aws/sensors/s3.py
Comment (pankajkoti) [inline astronomer/providers/amazon/aws/sensors/s3.py:20]:
```suggestion
                "This module is deprecated."
```

**Review by pankajkoti — APPROVED** [review]
LGTM once above review comments are addressed.

**Pull request merged**
